apputils/config_manager.py

```diff
@@ -0,0 +1,83 @@
+                        if os.path.isfile(os.path.join(config_path, f))]
+        for target in sorted(files_in_dir):
+            print("ConfigManager: Parsing config file: %s" % target)
+            config = self.get_scheduled_task_config_from_file(target)
+            if ConfigValidator.validate_config(config):
+                task_name = config["task_config"]["task_name"]
+                self.scheduled_tasks[task_name] = config
+                # Coerce `read_only` to boolean
+                if self.scheduled_tasks[task_name]["task_config"]["read_only"] == "true":  # NOQA
+                    self.scheduled_tasks[task_name]["task_config"]["read_only"] = True  # NOQA
+                else:
+                    self.scheduled_tasks[task_name]["task_config"]["read_only"] = False  # NOQA
+        self.print_tasks(self.scheduled_tasks)
+
+    def beat_tasks_from_config(self):
+        """Return dictionary that describes celerybeat tasks, from config."""
+        beats = {}
+        for j in self.scheduled_tasks.items():
+            beats[j[0]] = {'task': 'halocelery.tasks.generic_bound_containerized_task',  # NOQA
+                           'schedule': crontab(hour=j[1]["schedule"]["hour"],
+                                               minute=j[1]["schedule"]["minute"],  # NOQA
+                                               day_of_week=j[1]["schedule"]["day_of_week"],  # NOQA
+                                               day_of_month=j[1]["schedule"]["day_of_month"],  # NOQA
+                                               month_of_year=j[1]["schedule"]["month_of_year"]),  # NOQA
+                           'args': (j[1]["task_config"]["image"],
+                                    j[1]["env_literal"],
+                                    j[1]["env_expand"],
+                                    j[1]["task_config"]["retry"],
+                                    j[1]["log_config"],
+                                    j[1]["task_config"]["read_only"])}
+        return beats
+
+    @classmethod
+    def format_task(cls, task):
+        """Format task config into printable text."""
+        s_sched = "Schedule:\n\tMinute: $minute\n\tHour: $hour\n\tDay of week: $day_of_week\n\tDay of month: $day_of_month\n\tMonth of year: $month_of_year"  # NOQA
+        s_task = "Task: $task_name\nContainer image: $image\nRetries: $retry\n"
+        task_text = Template(s_task).safe_substitute(task["task_config"])
+        task_text += Template(s_sched).safe_substitute(task["schedule"])
+        return task_text
+
+    @classmethod
+    def get_scheduled_task_config_from_file(cls, config_file_path):
+        """Get scheduled task config from file.
+
+        Args:
+            config_file_path(str): Path to config file.
+
+        """
+        config = ConfigParser.RawConfigParser({}, dict)
+        config.optionxform = str
+        with open(config_file_path, 'r') as conf_file:
+            config.readfp(conf_file)
```

Comment:
> readfp(fp, filename=None)
> Deprecated since version 3.2: Use read_file() instead.
>

Reply:
@mong2 that function is not available in Python2.7.